eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxo.hs

```diff
@@ -233,24 +233,32 @@ validateCollateralContainsNonADA ::
   Map.Map (TxIn (EraCrypto era)) (TxOut era) ->
   Test (AlonzoUtxoPredFailure era)
 validateCollateralContainsNonADA txBody utxoCollateral =
-  -- When we do not have any non-ada TxOuts we can short-circuit the more
-  -- expensive validation, which has to compute the full balance on the Value,
-  -- not just the Coin.
-  if areAllAdaOnly utxoCollateral && areAllAdaOnly (txBody ^. collateralReturnTxBodyL)
-    then pure ()
-    else failureUnless (Val.isAdaOnly bal) $
-      case txBody ^. collateralReturnTxBodyL of
-        SJust retTxOut
-          | not (Val.isAdaOnly colbal) ->
-              CollateralContainsNonADA (retTxOut ^. valueTxOutL)
-        _ -> CollateralContainsNonADA colbal
+  -- When we do not have any non-ada TxOuts we can short-circuit the more expensive
```

Comment:
"We do not have any non-ada TxOuts"  means we have ada-only TxOuts, correct?  Or maybe: when we don't have multiassets?  Might be easier to understand without the double-negative, unless it's expressing something in particular